lib/ex_doc/formatter/html/autolink.ex

```diff
@@ -285,7 +285,17 @@ defmodule ExDoc.Formatter.HTML.Autolink do
 
   defp type_remote_url(source, alias, name, args) do
     name = enc("#{name}")
-    "#{source}#{enc(inspect(alias))}.html#t:#{name}/#{length(args)}"
+
+    if erlang_alias?(alias) do
+      "#{source}#{enc(to_string(alias))}.html#type-#{name}"
+    else
+      "#{source}#{enc(inspect(alias))}.html#t:#{name}/#{length(args)}"
+    end
+  end
+
+  defp erlang_alias?(alias) do
```

Comment:
I found ex_doc uses compiled regex - so probably we may use regex here? Any feedback?

Reply:
I think this is fine as is.  Can you please add a test?

Reply:
Added the test